test/extended/cpu_partitioning/crio.go

```diff
@@ -161,62 +161,65 @@ var _ = g.Describe("[sig-node][apigroup:config.openshift.io] CPU Partitioning no
-			// Collect the container data from the node
-			crioData, err := collectContainerInfo(ctx, oc, node, isClusterCPUPartitioned)
-			o.Expect(err).ToNot(o.HaveOccurred(), fmt.Sprintf("error getting crio container data from node %s", node.Name))
-
-			// When the cluster is CPU Partitioned there should always be pods with workload annotations. However,
-			// we do not enforce this check on non partitioned clusters since we don't expect nodes beyond masters to have pods
-			// that contain workload annotations. In the event that they do, we allow the logic to continue to validate
-			// CPUSets.
-			if isClusterCPUPartitioned {
-				o.Expect(crioData).ToNot(o.BeEmpty(), "error getting crio container data, container data should not be empty")
-			}
-
-			// CPU Core counts are numbers 0-(n-1) we iterate and create a CPUSet to compare.
-			// Note: CPU Value should be whole core counts, if situations come up where that's not the case
-			// we will need to revisit this approach.
-			totalCoreCount := []int{}
-			for x := 0; x < int(node.Status.Capacity.Cpu().Value()); x++ {
-				totalCoreCount = append(totalCoreCount, x)
-			}
-			parsedFullNodeCPUSet := cpuset.New(totalCoreCount...)
-
-			expectedCPUSet := cpuset.CPUSet{}
-
-			if _, isMaster := node.Labels[nodeMasterLabel]; isMaster {
-				expectedCPUSet = nodeRoleCPUSet.Master
-			} else if _, isWorker := node.Labels[nodeWorkerLabel]; isWorker {
-				expectedCPUSet = nodeRoleCPUSet.Worker
-			}
+			g.It(fmt.Sprintf("validating node %s for CPU affinity compliance", node.Name), func() {
+				// Collect the container data from the node
+				crioData, err := collectContainerInfo(ctx, oc, node, isClusterCPUPartitioned)
+				o.Expect(err).ToNot(o.HaveOccurred(), fmt.Sprintf("error getting crio container data from node %s", node.Name))
+
+				// When the cluster is CPU Partitioned there should always be pods with workload annotations. However,
+				// we do not enforce this check on non partitioned clusters since we don't expect nodes beyond masters to have pods
+				// that contain workload annotations. In the event that they do, we allow the logic to continue to validate
+				// CPUSets.
+				if isClusterCPUPartitioned {
+					o.Expect(crioData).ToNot(o.BeEmpty(), "error getting crio container data, container data should not be empty")
+				}
 
-			for _, containerInfo := range crioData {
-				parsedContainer, err := cpuset.Parse(containerInfo.CPUSet)
-				o.Expect(err).ToNot(o.HaveOccurred(), fmt.Sprintf("error parsing crio container cpuset %s", containerInfo.CPUSet))
-				parsedHost, err := cpuset.Parse(containerInfo.HostCPUSet)
-				o.Expect(err).ToNot(o.HaveOccurred(), fmt.Sprintf("error parsing host cpuset %s", containerInfo.HostCPUSet))
+				// CPU Core counts are numbers 0-(n-1) we iterate and create a CPUSet to compare.
+				// Note: CPU Value should be whole core counts, if situations come up where that's not the case
+				// we will need to revisit this approach.
```

Comment:
That's a good point, in this instance the `.Value()` returns `int64`, I'm just casting it to `int` in order to easily feed it into the CPUSet parser `cpuset.New(totalCoreCount...)` since it only takes a variadic list of `int`.

Reply:
The LGTM from my side